src/modules/campaign-monitor/services/campaign-monitor.service.spec.ts

```diff
@@ -525,12 +552,27 @@ describe('CampaignMonitorService', () => {
 				.reply(201, {});
 
 			try {
-				await campaignMonitorService.updateNewsletterPreferences(
-					{
-						newsletter: true,
-					},
-					mockUserInfo
-				);
+				await campaignMonitorService.updateNewsletterPreferences(mockUserInfo, {
+					newsletter: true,
+				});
+			} catch (e) {
+				expect(e).toBeUndefined();
+			}
+		});
+
+		it('should succesfully update newsletterPrefferences when no preferences are given (sync on login)', async () => {
+			nock(mockConfigService.get('CAMPAIGN_MONITOR_API_ENDPOINT') as string)
+				.post(
+					`/${mockConfigService.get(
+						'CAMPAIGN_MONITOR_SUBSCRIBER_API_VERSION'
+					)}/${mockConfigService.get(
+						'CAMPAIGN_MONITOR_SUBSCRIBER_API_ENDPOINT'
+					)}/${mockConfigService.get('CAMPAIGN_MONITOR_OPTIN_LIST_HETARCHIEF')}.json`
+				)
+				.reply(201, {});
+
+			try {
+				await campaignMonitorService.updateNewsletterPreferences(mockUserInfo);
 			} catch (e) {
 				expect(e).toBeUndefined();
```

Comment:
This should be a jest fail() statement, since we always want to fail, even if e is undefined, since it would mean that the updateNewsletterPreferences still threw an error causing it to execute the catch statement

Reply:
We don't want this test to fail. When it fails, e is defined and the test failed.